roachtest: activerecord failed

**Note:** This build has runtime assertions enabled. If the same failure was hit in a run without assertions enabled, there should be a similar failure without this message. If there isn&#39;t one, then this failure is likely due to an assertion violation or (assertion) timeout.

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18008144?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18008144?buildTab=artifacts#/activerecord) on master @ [b3fec61ca90095c664f6432af864f18e9946f8bb](https://github.com/cockroachdb/cockroach/commits/b3fec61ca90095c664f6432af864f18e9946f8bb):


```
--- FAIL: SecureTokenTest#test_token_value_not_overwritten_when_present (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: SecureTokenTest#test_regenerating_the_secure_token (unexpected)
--- FAIL: SecureTokenTest#test_token_calls_the_setter_method (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_embedded_data_is_different (unexpected)
--- FAIL: SuppressorTest#test_suppresses_update (unexpected)
--- FAIL: TokenForTest#test_finds_record_with_a_custom_primary_key (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_happens_only_once (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_nil (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_an_empty_string (unexpected)
--- FAIL: SuppressorTest#test_suppresses_validations_on_create (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_using_multiple_passwords (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_password (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_takes_the_same_amount_of_time_regardless_of_whether_record_is_found (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_does_not_authenticate_when_password_is_incorrect (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_subclass (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_is_skipped_if_column_was_not_selected (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_relation (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_specified_pk_and_sequence (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_has_expired (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_create_fixtures_resets_sequences_when_not_cached (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_attribute (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_when_password_is_correct (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_for_a_different_purpose (unexpected)
--- FAIL: TokenForTest#test_raises_when_no_primary_key_has_been_declared (unexpected)
--- FAIL: TokenForTest#test_raises_when_token_definition_does_not_exist (unexpected)
--- FAIL: TokenForTest#test_supports_JSON-serializable_embedded_data (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: SecurePasswordTest#test_authenticate_by_finds_record_using_multiple_attributes (unexpected)
--- FAIL: SuppressorTest#test_suppresses_create_in_callback (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_invalid (unexpected)
--- FAIL: SecureTokenTest#test_token_on_callback (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_accepts_any_object_that_implements_to_h (unexpected)
--- FAIL: SuppressorTest#test_resumes_saving_after_suppression_complete (unexpected)
--- FAIL: CockroachDB::FixturesResetPkSequenceTest#test_resets_to_min_pk_with_default_pk_and_sequence (unexpected)
--- FAIL: TokenForTest#test_finds_record_by_token (unexpected)
--- FAIL: TokenForTest#test_tokens_do_not_expire_by_default (unexpected)
--- FAIL: AutomaticInverseFindingTests#test_has_one_and_belongs_to_with_non_default_foreign_key_should_not_find_inverse_automatically (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_should_respect_table_alias (unexpected)
--- FAIL: TokenForTest#test_subclasses_can_redefine_tokens (unexpected)
--- FAIL: TokenForTest#test_finds_record_when_token_has_not_expired_and_embedded_data_has_not_changed (unexpected)
--- FAIL: TokenForTest#test_returns_nil_when_record_is_not_found (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_expires_in_is_different (unexpected)
--- FAIL: SecureTokenTest#test_token_values_are_generated_for_specified_attributes_and_persisted_on_save (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_double_belongs_to_destroys_both_from_either_end (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_association_name_is_the_same_as_join_table_name (unexpected)
--- FAIL: TokenForTest#test_raises_on_bang_when_record_is_not_found (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicLeases=default</code>
 - <code>runtimeAssertionsBuild=true</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-18008144/activerecord/1733208287620/1733215785473)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #136460 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.2 release-blocker]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-45127

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Roachtest_Openmetrics_Gce/18008097?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Roachtest_Openmetrics_Gce/18008097?buildTab=artifacts#/activerecord) on master @ [b3fec61ca90095c664f6432af864f18e9946f8bb](https://github.com/cockroachdb/cockroach/commits/b3fec61ca90095c664f6432af864f18e9946f8bb):


```
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: SuppressorTest#test_suppresses_create_in_callback (unexpected)
--- FAIL: TokenForTest#test_raises_when_token_definition_does_not_exist (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_association_name_is_the_same_as_join_table_name (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_does_not_authenticate_when_password_is_incorrect (unexpected)
--- FAIL: TokenForTest#test_supports_JSON-serializable_embedded_data (unexpected)
--- FAIL: TokenForTest#test_finds_record_by_token (unexpected)
--- FAIL: SuppressorTest#test_suppresses_update (unexpected)
--- FAIL: TokenForTest#test_tokens_do_not_expire_by_default (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_should_respect_table_alias (unexpected)
--- FAIL: SecureTokenTest#test_token_on_callback (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_for_a_different_purpose (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_happens_only_once (unexpected)
--- FAIL: SuppressorTest#test_suppresses_validations_on_create (unexpected)
--- FAIL: SecureTokenTest#test_token_values_are_generated_for_specified_attributes_and_persisted_on_save (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_an_empty_string (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_has_expired (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_using_multiple_passwords (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_password (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: SecurePasswordTest#test_authenticate_by_finds_record_using_multiple_attributes (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_takes_the_same_amount_of_time_regardless_of_whether_record_is_found (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_accepts_any_object_that_implements_to_h (unexpected)
--- FAIL: SecureTokenTest#test_token_calls_the_setter_method (unexpected)
--- FAIL: TokenForTest#test_finds_record_with_a_custom_primary_key (unexpected)
--- FAIL: AutomaticInverseFindingTests#test_has_one_and_belongs_to_with_non_default_foreign_key_should_not_find_inverse_automatically (unexpected)
--- FAIL: TokenForTest#test_finds_record_when_token_has_not_expired_and_embedded_data_has_not_changed (unexpected)
--- FAIL: SecureTokenTest#test_token_value_not_overwritten_when_present (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_attribute (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_nil (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_is_skipped_if_column_was_not_selected (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_double_belongs_to_destroys_both_from_either_end (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_embedded_data_is_different (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_invalid (unexpected)
--- FAIL: SuppressorTest#test_resumes_saving_after_suppression_complete (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_when_password_is_correct (unexpected)
--- FAIL: TokenForTest#test_raises_on_bang_when_record_is_not_found (unexpected)
--- FAIL: SecureTokenTest#test_regenerating_the_secure_token (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_relation (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_subclass (unexpected)
--- FAIL: TokenForTest#test_returns_nil_when_record_is_not_found (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_expires_in_is_different (unexpected)
--- FAIL: TokenForTest#test_subclasses_can_redefine_tokens (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: TokenForTest#test_raises_when_no_primary_key_has_been_declared (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicLeases=epoch</code>
 - <code>runtimeAssertionsBuild=false</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-18008097/activerecord/1733208330778/1733217265620)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #136562 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.3 release-blocker]
- #136561 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.3.1-rc release-blocker]
- #136460 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

**Note:** This build has runtime assertions enabled. If the same failure was hit in a run without assertions enabled, there should be a similar failure without this message. If there isn&#39;t one, then this failure is likely due to an assertion violation or (assertion) timeout.

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18024845?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18024845?buildTab=artifacts#/activerecord) on master @ [44de2d379610067e14a7ebfbc92e64311f13a232](https://github.com/cockroachdb/cockroach/commits/44de2d379610067e14a7ebfbc92e64311f13a232):


```
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_should_respect_table_alias (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_accepts_any_object_that_implements_to_h (unexpected)
--- FAIL: TokenForTest#test_subclasses_can_redefine_tokens (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_takes_the_same_amount_of_time_regardless_of_whether_record_is_found (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_when_password_is_correct (unexpected)
--- FAIL: SecureTokenTest#test_token_value_not_overwritten_when_present (unexpected)
--- FAIL: TokenForTest#test_raises_on_bang_when_record_is_not_found (unexpected)
--- FAIL: TokenForTest#test_supports_JSON-serializable_embedded_data (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_subclass (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_finds_record_using_multiple_attributes (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_using_multiple_passwords (unexpected)
--- FAIL: SecureTokenTest#test_token_calls_the_setter_method (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_relation (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_is_skipped_if_column_was_not_selected (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_has_expired (unexpected)
--- FAIL: TokenForTest#test_finds_record_by_token (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_an_empty_string (unexpected)
--- FAIL: TokenForTest#test_returns_nil_when_record_is_not_found (unexpected)
--- FAIL: TokenForTest#test_tokens_do_not_expire_by_default (unexpected)
--- FAIL: TokenForTest#test_finds_record_with_a_custom_primary_key (unexpected)
--- FAIL: SecureTokenTest#test_token_values_are_generated_for_specified_attributes_and_persisted_on_save (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_association_name_is_the_same_as_join_table_name (unexpected)
--- FAIL: SuppressorTest#test_suppresses_update (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_invalid (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_embedded_data_is_different (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_for_a_different_purpose (unexpected)
--- FAIL: SecureTokenTest#test_token_on_callback (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_nil (unexpected)
--- FAIL: SecureTokenTest#test_regenerating_the_secure_token (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_happens_only_once (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: HasOneAssociationsTest#test_has_one_double_belongs_to_destroys_both_from_either_end (unexpected)
--- FAIL: SuppressorTest#test_resumes_saving_after_suppression_complete (unexpected)
--- FAIL: SuppressorTest#test_suppresses_create_in_callback (unexpected)
--- FAIL: SuppressorTest#test_suppresses_validations_on_create (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_attribute (unexpected)
--- FAIL: TokenForTest#test_finds_record_when_token_has_not_expired_and_embedded_data_has_not_changed (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_does_not_authenticate_when_password_is_incorrect (unexpected)
--- FAIL: TokenForTest#test_raises_when_token_definition_does_not_exist (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_expires_in_is_different (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_password (unexpected)
--- FAIL: TokenForTest#test_raises_when_no_primary_key_has_been_declared (unexpected)
--- FAIL: AutomaticInverseFindingTests#test_has_one_and_belongs_to_with_non_default_foreign_key_should_not_find_inverse_automatically (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicLeases=default</code>
 - <code>runtimeAssertionsBuild=true</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-18024845/activerecord/1733295066306/1733302998705)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #136562 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.3 release-blocker]
- #136561 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.3.1-rc release-blocker]
- #136460 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Roachtest_Openmetrics_Gce/18025014?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Roachtest_Openmetrics_Gce/18025014?buildTab=artifacts#/activerecord) on master @ [44de2d379610067e14a7ebfbc92e64311f13a232](https://github.com/cockroachdb/cockroach/commits/44de2d379610067e14a7ebfbc92e64311f13a232):


```
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: AutomaticInverseFindingTests#test_has_one_and_belongs_to_with_non_default_foreign_key_should_not_find_inverse_automatically (unexpected)
--- FAIL: TokenForTest#test_tokens_do_not_expire_by_default (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_password (unexpected)
--- FAIL: SecureTokenTest#test_token_on_callback (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_happens_only_once (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_takes_the_same_amount_of_time_regardless_of_whether_record_is_found (unexpected)
--- FAIL: SecureTokenTest#test_token_value_not_overwritten_when_present (unexpected)
--- FAIL: TokenForTest#test_returns_nil_when_record_is_not_found (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_association_name_is_the_same_as_join_table_name (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_relation (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_an_empty_string (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_for_a_different_purpose (unexpected)
--- FAIL: TokenForTest#test_finds_record_by_token (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_expires_in_is_different (unexpected)
--- FAIL: TokenForTest#test_supports_JSON-serializable_embedded_data (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_invalid (unexpected)
--- FAIL: TokenForTest#test_subclasses_can_redefine_tokens (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_attribute (unexpected)
--- FAIL: SuppressorTest#test_suppresses_update (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_should_respect_table_alias (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_does_not_authenticate_when_password_is_incorrect (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_has_expired (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_finds_record_using_multiple_attributes (unexpected)
--- FAIL: SuppressorTest#test_suppresses_create_in_callback (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_embedded_data_is_different (unexpected)
--- FAIL: SecureTokenTest#test_token_calls_the_setter_method (unexpected)
--- FAIL: SuppressorTest#test_suppresses_validations_on_create (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_is_skipped_if_column_was_not_selected (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_double_belongs_to_destroys_both_from_either_end (unexpected)
--- FAIL: SecureTokenTest#test_token_values_are_generated_for_specified_attributes_and_persisted_on_save (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_nil (unexpected)
--- FAIL: TokenForTest#test_raises_when_token_definition_does_not_exist (unexpected)
--- FAIL: SuppressorTest#test_resumes_saving_after_suppression_complete (unexpected)
--- FAIL: TokenForTest#test_raises_on_bang_when_record_is_not_found (unexpected)
--- FAIL: TokenForTest#test_finds_record_when_token_has_not_expired_and_embedded_data_has_not_changed (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_using_multiple_passwords (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_subclass (unexpected)
--- FAIL: SecureTokenTest#test_regenerating_the_secure_token (unexpected)
--- FAIL: TokenForTest#test_raises_when_no_primary_key_has_been_declared (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_when_password_is_correct (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_accepts_any_object_that_implements_to_h (unexpected)
--- FAIL: TokenForTest#test_finds_record_with_a_custom_primary_key (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicLeases=epoch</code>
 - <code>runtimeAssertionsBuild=false</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-18025014/activerecord/1733295055427/1733303357020)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #136562 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.3 release-blocker]
- #136561 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.3.1-rc release-blocker]
- #136460 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Roachtest_Openmetrics_Gce/18028758?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Roachtest_Openmetrics_Gce/18028758?buildTab=artifacts#/activerecord) on master @ [d87d47386f47e826d4694fa50113a1d9ff6a9612](https://github.com/cockroachdb/cockroach/commits/d87d47386f47e826d4694fa50113a1d9ff6a9612):


```
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: SuppressorTest#test_suppresses_update (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_expires_in_is_different (unexpected)
--- FAIL: SecureTokenTest#test_regenerating_the_secure_token (unexpected)
--- FAIL: SecureTokenTest#test_token_on_callback (unexpected)
--- FAIL: SuppressorTest#test_suppresses_create_in_callback (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_embedded_data_is_different (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_has_expired (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_finds_record_using_multiple_attributes (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_association_name_is_the_same_as_join_table_name (unexpected)
--- FAIL: AutomaticInverseFindingTests#test_has_one_and_belongs_to_with_non_default_foreign_key_should_not_find_inverse_automatically (unexpected)
--- FAIL: SecureTokenTest#test_token_calls_the_setter_method (unexpected)
--- FAIL: SuppressorTest#test_suppresses_validations_on_create (unexpected)
--- FAIL: TokenForTest#test_tokens_do_not_expire_by_default (unexpected)
--- FAIL: TokenForTest#test_finds_record_when_token_has_not_expired_and_embedded_data_has_not_changed (unexpected)
--- FAIL: TokenForTest#test_raises_on_bang_when_record_is_not_found (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_takes_the_same_amount_of_time_regardless_of_whether_record_is_found (unexpected)
--- FAIL: TokenForTest#test_finds_record_with_a_custom_primary_key (unexpected)
--- FAIL: TokenForTest#test_raises_when_no_primary_key_has_been_declared (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_accepts_any_object_that_implements_to_h (unexpected)
--- FAIL: SecureTokenTest#test_token_value_not_overwritten_when_present (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_when_password_is_correct (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_an_empty_string (unexpected)
--- FAIL: TokenForTest#test_subclasses_can_redefine_tokens (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_double_belongs_to_destroys_both_from_either_end (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_should_respect_table_alias (unexpected)
--- FAIL: TokenForTest#test_raises_when_token_definition_does_not_exist (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_is_skipped_if_column_was_not_selected (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_relation (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_for_a_different_purpose (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_using_multiple_passwords (unexpected)
--- FAIL: TokenForTest#test_supports_JSON-serializable_embedded_data (unexpected)
--- FAIL: SecureTokenTest#test_token_values_are_generated_for_specified_attributes_and_persisted_on_save (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_does_not_authenticate_when_password_is_incorrect (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_happens_only_once (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_nil (unexpected)
--- FAIL: TokenForTest#test_returns_nil_when_record_is_not_found (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_password (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_subclass (unexpected)
--- FAIL: TokenForTest#test_finds_record_by_token (unexpected)
--- FAIL: SuppressorTest#test_resumes_saving_after_suppression_complete (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_attribute (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_invalid (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/cpu_arch=arm64/run_1
```

Parameters:
 - <code>arch=arm64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=false</code>
 - <code>metamorphicLeases=leader</code>
 - <code>runtimeAssertionsBuild=false</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-18028758/activerecord/1733316492913/1733327547074)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #136675 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.2.6-rc release-blocker]
- #136562 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3 release-blocker]
- #136561 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3.1-rc]
- #136460 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Roachtest_Openmetrics_Gce/18029618?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Roachtest_Openmetrics_Gce/18029618?buildTab=artifacts#/activerecord) on master @ [535f8f31b122261c54e0b27f803010c71b25377f](https://github.com/cockroachdb/cockroach/commits/535f8f31b122261c54e0b27f803010c71b25377f):


```
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: SecurePasswordTest#test_authenticate_by_finds_record_using_multiple_attributes (unexpected)
--- FAIL: TokenForTest#test_raises_on_bang_when_record_is_not_found (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_subclass (unexpected)
--- FAIL: TokenForTest#test_raises_when_token_definition_does_not_exist (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_has_expired (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_is_skipped_if_column_was_not_selected (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_password (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_happens_only_once (unexpected)
--- FAIL: SecureTokenTest#test_regenerating_the_secure_token (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_attribute (unexpected)
--- FAIL: SecureTokenTest#test_token_calls_the_setter_method (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_for_a_different_purpose (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_should_respect_table_alias (unexpected)
--- FAIL: SuppressorTest#test_resumes_saving_after_suppression_complete (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: SecureTokenTest#test_token_on_callback (unexpected)
--- FAIL: SecureTokenTest#test_token_values_are_generated_for_specified_attributes_and_persisted_on_save (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_takes_the_same_amount_of_time_regardless_of_whether_record_is_found (unexpected)
--- FAIL: TokenForTest#test_returns_nil_when_record_is_not_found (unexpected)
--- FAIL: TokenForTest#test_finds_record_with_a_custom_primary_key (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_an_empty_string (unexpected)
--- FAIL: TokenForTest#test_tokens_do_not_expire_by_default (unexpected)
--- FAIL: TokenForTest#test_finds_record_when_token_has_not_expired_and_embedded_data_has_not_changed (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_invalid (unexpected)
--- FAIL: TokenForTest#test_subclasses_can_redefine_tokens (unexpected)
--- FAIL: TokenForTest#test_finds_record_by_token (unexpected)
--- FAIL: SecureTokenTest#test_token_value_not_overwritten_when_present (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_nil (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_accepts_any_object_that_implements_to_h (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_does_not_authenticate_when_password_is_incorrect (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_expires_in_is_different (unexpected)
--- FAIL: TokenForTest#test_raises_when_no_primary_key_has_been_declared (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_using_multiple_passwords (unexpected)
--- FAIL: TokenForTest#test_supports_JSON-serializable_embedded_data (unexpected)
--- FAIL: AutomaticInverseFindingTests#test_has_one_and_belongs_to_with_non_default_foreign_key_should_not_find_inverse_automatically (unexpected)
--- FAIL: SuppressorTest#test_suppresses_create_in_callback (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_association_name_is_the_same_as_join_table_name (unexpected)
--- FAIL: SuppressorTest#test_suppresses_validations_on_create (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_double_belongs_to_destroys_both_from_either_end (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_relation (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_embedded_data_is_different (unexpected)
--- FAIL: SuppressorTest#test_suppresses_update (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_when_password_is_correct (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicLeases=default</code>
 - <code>runtimeAssertionsBuild=false</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-18029618/activerecord/1733320260720/1733327983854)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #136675 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.2.6-rc release-blocker]
- #136562 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3 release-blocker]
- #136561 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3.1-rc]
- #136460 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Roachtest_Openmetrics_Gce/18030459?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Roachtest_Openmetrics_Gce/18030459?buildTab=artifacts#/activerecord) on master @ [78e363f4cbd720292aad2a3bed24790cd1573fd2](https://github.com/cockroachdb/cockroach/commits/78e363f4cbd720292aad2a3bed24790cd1573fd2):


```
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: TokenForTest#test_finds_record_when_token_has_not_expired_and_embedded_data_has_not_changed (unexpected)
--- FAIL: TokenForTest#test_raises_when_no_primary_key_has_been_declared (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_association_name_is_the_same_as_join_table_name (unexpected)
--- FAIL: SecureTokenTest#test_token_value_not_overwritten_when_present (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_attribute (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_expires_in_is_different (unexpected)
--- FAIL: TokenForTest#test_finds_record_with_a_custom_primary_key (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_does_not_authenticate_when_password_is_incorrect (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_when_password_is_correct (unexpected)
--- FAIL: TokenForTest#test_raises_on_bang_when_record_is_not_found (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_should_respect_table_alias (unexpected)
--- FAIL: SuppressorTest#test_suppresses_create_in_callback (unexpected)
--- FAIL: SuppressorTest#test_suppresses_validations_on_create (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_subclass (unexpected)
--- FAIL: TokenForTest#test_tokens_do_not_expire_by_default (unexpected)
--- FAIL: AutomaticInverseFindingTests#test_has_one_and_belongs_to_with_non_default_foreign_key_should_not_find_inverse_automatically (unexpected)
--- FAIL: SecureTokenTest#test_token_on_callback (unexpected)
--- FAIL: TokenForTest#test_returns_nil_when_record_is_not_found (unexpected)
--- FAIL: SecureTokenTest#test_token_calls_the_setter_method (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_an_empty_string (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_nil (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_has_expired (unexpected)
--- FAIL: SuppressorTest#test_resumes_saving_after_suppression_complete (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_invalid (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_embedded_data_is_different (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_accepts_any_object_that_implements_to_h (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_takes_the_same_amount_of_time_regardless_of_whether_record_is_found (unexpected)
--- FAIL: SecureTokenTest#test_regenerating_the_secure_token (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_relation (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_using_multiple_passwords (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_is_skipped_if_column_was_not_selected (unexpected)
--- FAIL: TokenForTest#test_raises_when_token_definition_does_not_exist (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_password (unexpected)
--- FAIL: TokenForTest#test_finds_record_by_token (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: SuppressorTest#test_suppresses_update (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_finds_record_using_multiple_attributes (unexpected)
--- FAIL: TokenForTest#test_subclasses_can_redefine_tokens (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_double_belongs_to_destroys_both_from_either_end (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_for_a_different_purpose (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_happens_only_once (unexpected)
--- FAIL: TokenForTest#test_supports_JSON-serializable_embedded_data (unexpected)
--- FAIL: SecureTokenTest#test_token_values_are_generated_for_specified_attributes_and_persisted_on_save (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicLeases=epoch</code>
 - <code>runtimeAssertionsBuild=false</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-18030459/activerecord/1733323594584/1733332011679)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #136675 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.2.6-rc release-blocker]
- #136562 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3 release-blocker]
- #136561 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3.1-rc]
- #136460 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Roachtest_Openmetrics_Gce/18031328?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Roachtest_Openmetrics_Gce/18031328?buildTab=artifacts#/activerecord) on master @ [757b9ab8278cbbfb9ce468341e71a3611d7857b2](https://github.com/cockroachdb/cockroach/commits/757b9ab8278cbbfb9ce468341e71a3611d7857b2):


```
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: SecureTokenTest#test_token_values_are_generated_for_specified_attributes_and_persisted_on_save (unexpected)
--- FAIL: TokenForTest#test_finds_record_by_token (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_expires_in_is_different (unexpected)
--- FAIL: TokenForTest#test_raises_on_bang_when_record_is_not_found (unexpected)
--- FAIL: AutomaticInverseFindingTests#test_has_one_and_belongs_to_with_non_default_foreign_key_should_not_find_inverse_automatically (unexpected)
--- FAIL: TokenForTest#test_raises_when_token_definition_does_not_exist (unexpected)
--- FAIL: SecureTokenTest#test_regenerating_the_secure_token (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_when_password_is_correct (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_is_skipped_if_column_was_not_selected (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_nil (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_relation (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_finds_record_using_multiple_attributes (unexpected)
--- FAIL: TokenForTest#test_finds_record_when_token_has_not_expired_and_embedded_data_has_not_changed (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_embedded_data_is_different (unexpected)
--- FAIL: TokenForTest#test_raises_when_no_primary_key_has_been_declared (unexpected)
--- FAIL: TokenForTest#test_returns_nil_when_record_is_not_found (unexpected)
--- FAIL: TokenForTest#test_tokens_do_not_expire_by_default (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_association_name_is_the_same_as_join_table_name (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_password (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_double_belongs_to_destroys_both_from_either_end (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_accepts_any_object_that_implements_to_h (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_does_not_authenticate_when_password_is_incorrect (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_should_respect_table_alias (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_for_a_different_purpose (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_subclass (unexpected)
--- FAIL: SecureTokenTest#test_token_value_not_overwritten_when_present (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_happens_only_once (unexpected)
--- FAIL: TokenForTest#test_supports_JSON-serializable_embedded_data (unexpected)
--- FAIL: TokenForTest#test_finds_record_with_a_custom_primary_key (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_an_empty_string (unexpected)
--- FAIL: SecureTokenTest#test_token_on_callback (unexpected)
--- FAIL: SuppressorTest#test_suppresses_create_in_callback (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_takes_the_same_amount_of_time_regardless_of_whether_record_is_found (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: SuppressorTest#test_suppresses_validations_on_create (unexpected)
--- FAIL: SuppressorTest#test_suppresses_update (unexpected)
--- FAIL: SecureTokenTest#test_token_calls_the_setter_method (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_has_expired (unexpected)
--- FAIL: SuppressorTest#test_resumes_saving_after_suppression_complete (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_using_multiple_passwords (unexpected)
--- FAIL: TokenForTest#test_subclasses_can_redefine_tokens (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_invalid (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_attribute (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicLeases=expiration</code>
 - <code>runtimeAssertionsBuild=false</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-18031328/activerecord/1733327303228/1733335670829)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #136675 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.2.6-rc release-blocker]
- #136562 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3 release-blocker]
- #136561 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3.1-rc]
- #136460 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Roachtest_Openmetrics_Gce/18034978?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Roachtest_Openmetrics_Gce/18034978?buildTab=artifacts#/activerecord) on master @ [826681a388c27524b3b3c5a5b66aaadafd05fc40](https://github.com/cockroachdb/cockroach/commits/826681a388c27524b3b3c5a5b66aaadafd05fc40):


```
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_attribute (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_for_a_different_purpose (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_embedded_data_is_different (unexpected)
--- FAIL: TokenForTest#test_returns_nil_when_record_is_not_found (unexpected)
--- FAIL: TokenForTest#test_raises_when_no_primary_key_has_been_declared (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_password (unexpected)
--- FAIL: SuppressorTest#test_resumes_saving_after_suppression_complete (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: SecurePasswordTest#test_authenticate_by_takes_the_same_amount_of_time_regardless_of_whether_record_is_found (unexpected)
--- FAIL: SecureTokenTest#test_regenerating_the_secure_token (unexpected)
--- FAIL: TokenForTest#test_subclasses_can_redefine_tokens (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_expires_in_is_different (unexpected)
--- FAIL: SecureTokenTest#test_token_value_not_overwritten_when_present (unexpected)
--- FAIL: SuppressorTest#test_suppresses_validations_on_create (unexpected)
--- FAIL: SecureTokenTest#test_token_on_callback (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_association_name_is_the_same_as_join_table_name (unexpected)
--- FAIL: SuppressorTest#test_suppresses_create_in_callback (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_finds_record_using_multiple_attributes (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_accepts_any_object_that_implements_to_h (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: SecureTokenTest#test_token_values_are_generated_for_specified_attributes_and_persisted_on_save (unexpected)
--- FAIL: TokenForTest#test_tokens_do_not_expire_by_default (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_when_password_is_correct (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_using_multiple_passwords (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_subclass (unexpected)
--- FAIL: SuppressorTest#test_suppresses_update (unexpected)
--- FAIL: TokenForTest#test_supports_JSON-serializable_embedded_data (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_nil (unexpected)
--- FAIL: TokenForTest#test_finds_record_with_a_custom_primary_key (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_has_expired (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_invalid (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_is_skipped_if_column_was_not_selected (unexpected)
--- FAIL: TokenForTest#test_raises_on_bang_when_record_is_not_found (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_relation (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_happens_only_once (unexpected)
--- FAIL: TokenForTest#test_finds_record_when_token_has_not_expired_and_embedded_data_has_not_changed (unexpected)
--- FAIL: TokenForTest#test_raises_when_token_definition_does_not_exist (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_does_not_authenticate_when_password_is_incorrect (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_an_empty_string (unexpected)
--- FAIL: TokenForTest#test_finds_record_by_token (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_double_belongs_to_destroys_both_from_either_end (unexpected)
--- FAIL: SecureTokenTest#test_token_calls_the_setter_method (unexpected)
--- FAIL: AutomaticInverseFindingTests#test_has_one_and_belongs_to_with_non_default_foreign_key_should_not_find_inverse_automatically (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_should_respect_table_alias (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicLeases=leader</code>
 - <code>runtimeAssertionsBuild=false</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-18034978/activerecord/1733338299633/1733347139843)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #136675 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.2.6-rc]
- #136562 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3 release-blocker]
- #136561 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3.1-rc]
- #136460 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Roachtest_Openmetrics_Gce/18036065?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Roachtest_Openmetrics_Gce/18036065?buildTab=artifacts#/activerecord) on master @ [74e821f1d1ce358f49354f8581b8663aa4c2fb15](https://github.com/cockroachdb/cockroach/commits/74e821f1d1ce358f49354f8581b8663aa4c2fb15):


```
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: TokenForTest#test_finds_record_with_a_custom_primary_key (unexpected)
--- FAIL: TokenForTest#test_subclasses_can_redefine_tokens (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_for_a_different_purpose (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_happens_only_once (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_accepts_any_object_that_implements_to_h (unexpected)
--- FAIL: TokenForTest#test_tokens_do_not_expire_by_default (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_finds_record_using_multiple_attributes (unexpected)
--- FAIL: TokenForTest#test_supports_JSON-serializable_embedded_data (unexpected)
--- FAIL: TokenForTest#test_raises_on_bang_when_record_is_not_found (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: HasOneAssociationsTest#test_has_one_double_belongs_to_destroys_both_from_either_end (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_relation (unexpected)
--- FAIL: TokenForTest#test_finds_record_when_token_has_not_expired_and_embedded_data_has_not_changed (unexpected)
--- FAIL: SecureTokenTest#test_token_value_not_overwritten_when_present (unexpected)
--- FAIL: SecureTokenTest#test_token_values_are_generated_for_specified_attributes_and_persisted_on_save (unexpected)
--- FAIL: TokenForTest#test_raises_when_no_primary_key_has_been_declared (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_is_skipped_if_column_was_not_selected (unexpected)
--- FAIL: TokenForTest#test_raises_when_token_definition_does_not_exist (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_does_not_authenticate_when_password_is_incorrect (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_embedded_data_is_different (unexpected)
--- FAIL: SuppressorTest#test_suppresses_update (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_invalid (unexpected)
--- FAIL: TokenForTest#test_finds_record_by_token (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_should_respect_table_alias (unexpected)
--- FAIL: TokenForTest#test_returns_nil_when_record_is_not_found (unexpected)
--- FAIL: AutomaticInverseFindingTests#test_has_one_and_belongs_to_with_non_default_foreign_key_should_not_find_inverse_automatically (unexpected)
--- FAIL: SuppressorTest#test_resumes_saving_after_suppression_complete (unexpected)
--- FAIL: SecureTokenTest#test_regenerating_the_secure_token (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_subclass (unexpected)
--- FAIL: SecureTokenTest#test_token_calls_the_setter_method (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_expires_in_is_different (unexpected)
--- FAIL: SuppressorTest#test_suppresses_create_in_callback (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_password (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_nil (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_takes_the_same_amount_of_time_regardless_of_whether_record_is_found (unexpected)
--- FAIL: SuppressorTest#test_suppresses_validations_on_create (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_using_multiple_passwords (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_an_empty_string (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_when_password_is_correct (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_has_expired (unexpected)
--- FAIL: SecureTokenTest#test_token_on_callback (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_association_name_is_the_same_as_join_table_name (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_attribute (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicLeases=leader</code>
 - <code>runtimeAssertionsBuild=false</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-18036065/activerecord/1733342121578/1733351526992)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #136675 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.2.6-rc]
- #136562 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3 release-blocker]
- #136561 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3.1-rc]
- #136460 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Roachtest_Openmetrics_Gce/18036970?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Roachtest_Openmetrics_Gce/18036970?buildTab=artifacts#/activerecord) on master @ [5a7850a72f941992b1bb4b23a73b5fa5e9f15a68](https://github.com/cockroachdb/cockroach/commits/5a7850a72f941992b1bb4b23a73b5fa5e9f15a68):


```
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_should_respect_table_alias (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_takes_the_same_amount_of_time_regardless_of_whether_record_is_found (unexpected)
--- FAIL: TokenForTest#test_raises_on_bang_when_record_is_not_found (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: TokenForTest#test_subclasses_can_redefine_tokens (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_an_empty_string (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_happens_only_once (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_accepts_any_object_that_implements_to_h (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_for_a_different_purpose (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_expires_in_is_different (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_has_expired (unexpected)
--- FAIL: TokenForTest#test_tokens_do_not_expire_by_default (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_finds_record_using_multiple_attributes (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_attribute (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: HasOneAssociationsTest#test_has_one_double_belongs_to_destroys_both_from_either_end (unexpected)
--- FAIL: SecureTokenTest#test_token_values_are_generated_for_specified_attributes_and_persisted_on_save (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_using_multiple_passwords (unexpected)
--- FAIL: AutomaticInverseFindingTests#test_has_one_and_belongs_to_with_non_default_foreign_key_should_not_find_inverse_automatically (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_does_not_authenticate_when_password_is_incorrect (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_password (unexpected)
--- FAIL: TokenForTest#test_returns_nil_when_record_is_not_found (unexpected)
--- FAIL: SuppressorTest#test_suppresses_validations_on_create (unexpected)
--- FAIL: SecureTokenTest#test_regenerating_the_secure_token (unexpected)
--- FAIL: SecureTokenTest#test_token_calls_the_setter_method (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_relation (unexpected)
--- FAIL: TokenForTest#test_supports_JSON-serializable_embedded_data (unexpected)
--- FAIL: TokenForTest#test_raises_when_token_definition_does_not_exist (unexpected)
--- FAIL: SuppressorTest#test_suppresses_update (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_when_password_is_correct (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_nil (unexpected)
--- FAIL: SuppressorTest#test_suppresses_create_in_callback (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_subclass (unexpected)
--- FAIL: TokenForTest#test_finds_record_with_a_custom_primary_key (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_embedded_data_is_different (unexpected)
--- FAIL: TokenForTest#test_finds_record_when_token_has_not_expired_and_embedded_data_has_not_changed (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_association_name_is_the_same_as_join_table_name (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_invalid (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_is_skipped_if_column_was_not_selected (unexpected)
--- FAIL: SecureTokenTest#test_token_value_not_overwritten_when_present (unexpected)
--- FAIL: TokenForTest#test_finds_record_by_token (unexpected)
--- FAIL: TokenForTest#test_raises_when_no_primary_key_has_been_declared (unexpected)
--- FAIL: SecureTokenTest#test_token_on_callback (unexpected)
--- FAIL: SuppressorTest#test_resumes_saving_after_suppression_complete (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicLeases=expiration</code>
 - <code>runtimeAssertionsBuild=false</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-18036970/activerecord/1733345208750/1733352901768)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #136675 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.2.6-rc]
- #136562 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3 release-blocker]
- #136561 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3.1-rc]
- #136460 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Roachtest_Openmetrics_Gce/18038658?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Roachtest_Openmetrics_Gce/18038658?buildTab=artifacts#/activerecord) on master @ [b4280f7c6613f362791478696adbdf2cc67dd874](https://github.com/cockroachdb/cockroach/commits/b4280f7c6613f362791478696adbdf2cc67dd874):


```
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: AutomaticInverseFindingTests#test_has_one_and_belongs_to_with_non_default_foreign_key_should_not_find_inverse_automatically (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_invalid (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_happens_only_once (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_for_a_different_purpose (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_accepts_any_object_that_implements_to_h (unexpected)
--- FAIL: TokenForTest#test_raises_when_token_definition_does_not_exist (unexpected)
--- FAIL: TokenForTest#test_subclasses_can_redefine_tokens (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_an_empty_string (unexpected)
--- FAIL: SuppressorTest#test_suppresses_validations_on_create (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_password (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_should_respect_table_alias (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_is_skipped_if_column_was_not_selected (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_double_belongs_to_destroys_both_from_either_end (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_finds_record_using_multiple_attributes (unexpected)
--- FAIL: SuppressorTest#test_suppresses_create_in_callback (unexpected)
--- FAIL: TokenForTest#test_finds_record_with_a_custom_primary_key (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_does_not_authenticate_when_password_is_incorrect (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_attribute (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_nil (unexpected)
--- FAIL: TokenForTest#test_raises_when_no_primary_key_has_been_declared (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_subclass (unexpected)
--- FAIL: SecureTokenTest#test_token_calls_the_setter_method (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_when_password_is_correct (unexpected)
--- FAIL: TokenForTest#test_finds_record_when_token_has_not_expired_and_embedded_data_has_not_changed (unexpected)
--- FAIL: TokenForTest#test_supports_JSON-serializable_embedded_data (unexpected)
--- FAIL: SuppressorTest#test_suppresses_update (unexpected)
--- FAIL: SecureTokenTest#test_token_values_are_generated_for_specified_attributes_and_persisted_on_save (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_using_multiple_passwords (unexpected)
--- FAIL: SuppressorTest#test_resumes_saving_after_suppression_complete (unexpected)
--- FAIL: TokenForTest#test_tokens_do_not_expire_by_default (unexpected)
--- FAIL: TokenForTest#test_returns_nil_when_record_is_not_found (unexpected)
--- FAIL: SecureTokenTest#test_regenerating_the_secure_token (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_relation (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_association_name_is_the_same_as_join_table_name (unexpected)
--- FAIL: SecureTokenTest#test_token_value_not_overwritten_when_present (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_embedded_data_is_different (unexpected)
--- FAIL: TokenForTest#test_raises_on_bang_when_record_is_not_found (unexpected)
--- FAIL: SecureTokenTest#test_token_on_callback (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_has_expired (unexpected)
--- FAIL: TokenForTest#test_finds_record_by_token (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_expires_in_is_different (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_takes_the_same_amount_of_time_regardless_of_whether_record_is_found (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicLeases=default</code>
 - <code>runtimeAssertionsBuild=false</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-18038658/activerecord/1733349094033/1733356742500)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #136675 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.2.6-rc]
- #136562 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3 release-blocker]
- #136561 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3.1-rc]
- #136460 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Roachtest_Openmetrics_Gce/18040227?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Roachtest_Openmetrics_Gce/18040227?buildTab=artifacts#/activerecord) on master @ [0500ce6dc5974af55867c6da0c816e0e64e472b0](https://github.com/cockroachdb/cockroach/commits/0500ce6dc5974af55867c6da0c816e0e64e472b0):


```
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_password (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_has_expired (unexpected)
--- FAIL: TokenForTest#test_supports_JSON-serializable_embedded_data (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_is_skipped_if_column_was_not_selected (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_expires_in_is_different (unexpected)
--- FAIL: TokenForTest#test_tokens_do_not_expire_by_default (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_happens_only_once (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_when_password_is_correct (unexpected)
--- FAIL: TokenForTest#test_raises_when_token_definition_does_not_exist (unexpected)
--- FAIL: SecureTokenTest#test_token_on_callback (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_relation (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_accepts_any_object_that_implements_to_h (unexpected)
--- FAIL: SecureTokenTest#test_token_value_not_overwritten_when_present (unexpected)
--- FAIL: TokenForTest#test_finds_record_with_a_custom_primary_key (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_invalid (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_attribute (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_using_multiple_passwords (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_double_belongs_to_destroys_both_from_either_end (unexpected)
--- FAIL: TokenForTest#test_subclasses_can_redefine_tokens (unexpected)
--- FAIL: TokenForTest#test_raises_on_bang_when_record_is_not_found (unexpected)
--- FAIL: TokenForTest#test_finds_record_by_token (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_finds_record_using_multiple_attributes (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_for_a_different_purpose (unexpected)
--- FAIL: SuppressorTest#test_suppresses_create_in_callback (unexpected)
--- FAIL: TokenForTest#test_raises_when_no_primary_key_has_been_declared (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_subclass (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_association_name_is_the_same_as_join_table_name (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_embedded_data_is_different (unexpected)
--- FAIL: TokenForTest#test_returns_nil_when_record_is_not_found (unexpected)
--- FAIL: SuppressorTest#test_suppresses_update (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_an_empty_string (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_nil (unexpected)
--- FAIL: SuppressorTest#test_resumes_saving_after_suppression_complete (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_should_respect_table_alias (unexpected)
--- FAIL: SecureTokenTest#test_regenerating_the_secure_token (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_takes_the_same_amount_of_time_regardless_of_whether_record_is_found (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_does_not_authenticate_when_password_is_incorrect (unexpected)
--- FAIL: SecureTokenTest#test_token_calls_the_setter_method (unexpected)
--- FAIL: AutomaticInverseFindingTests#test_has_one_and_belongs_to_with_non_default_foreign_key_should_not_find_inverse_automatically (unexpected)
--- FAIL: SuppressorTest#test_suppresses_validations_on_create (unexpected)
--- FAIL: TokenForTest#test_finds_record_when_token_has_not_expired_and_embedded_data_has_not_changed (unexpected)
--- FAIL: SecureTokenTest#test_token_values_are_generated_for_specified_attributes_and_persisted_on_save (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicLeases=default</code>
 - <code>runtimeAssertionsBuild=false</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-18040227/activerecord/1733356144720/1733364729368)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #136675 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.2.6-rc]
- #136562 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3 release-blocker]
- #136561 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3.1-rc]
- #136460 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18041104?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18041104?buildTab=artifacts#/activerecord) on master @ [844d7637f3b4dc1275e8aa05c1cf3bbb1f59f8eb](https://github.com/cockroachdb/cockroach/commits/844d7637f3b4dc1275e8aa05c1cf3bbb1f59f8eb):


```
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: TokenForTest#test_does_not_find_record_when_token_has_expired (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_happens_only_once (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_expires_in_is_different (unexpected)
--- FAIL: TokenForTest#test_subclasses_can_redefine_tokens (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_finds_record_using_multiple_attributes (unexpected)
--- FAIL: TokenForTest#test_raises_when_no_primary_key_has_been_declared (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_double_belongs_to_destroys_both_from_either_end (unexpected)
--- FAIL: SuppressorTest#test_suppresses_update (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_does_not_authenticate_when_password_is_incorrect (unexpected)
--- FAIL: TokenForTest#test_supports_JSON-serializable_embedded_data (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_should_respect_table_alias (unexpected)
--- FAIL: SecureTokenTest#test_token_on_callback (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_for_a_different_purpose (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_embedded_data_is_different (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_accepts_any_object_that_implements_to_h (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: TokenForTest#test_raises_on_bang_when_record_is_not_found (unexpected)
--- FAIL: SecureTokenTest#test_token_value_not_overwritten_when_present (unexpected)
--- FAIL: SecureTokenTest#test_token_values_are_generated_for_specified_attributes_and_persisted_on_save (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_attribute (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_relation (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_takes_the_same_amount_of_time_regardless_of_whether_record_is_found (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_when_password_is_correct (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_password (unexpected)
--- FAIL: SecureTokenTest#test_regenerating_the_secure_token (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_using_multiple_passwords (unexpected)
--- FAIL: SuppressorTest#test_resumes_saving_after_suppression_complete (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_is_skipped_if_column_was_not_selected (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: SecureTokenTest#test_token_calls_the_setter_method (unexpected)
--- FAIL: SuppressorTest#test_suppresses_create_in_callback (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_subclass (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_association_name_is_the_same_as_join_table_name (unexpected)
--- FAIL: TokenForTest#test_finds_record_with_a_custom_primary_key (unexpected)
--- FAIL: TokenForTest#test_tokens_do_not_expire_by_default (unexpected)
--- FAIL: SuppressorTest#test_suppresses_validations_on_create (unexpected)
--- FAIL: TokenForTest#test_returns_nil_when_record_is_not_found (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_an_empty_string (unexpected)
--- FAIL: TokenForTest#test_finds_record_by_token (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_nil (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_invalid (unexpected)
--- FAIL: AutomaticInverseFindingTests#test_has_one_and_belongs_to_with_non_default_foreign_key_should_not_find_inverse_automatically (unexpected)
--- FAIL: TokenForTest#test_finds_record_when_token_has_not_expired_and_embedded_data_has_not_changed (unexpected)
--- FAIL: TokenForTest#test_raises_when_token_definition_does_not_exist (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicLeases=expiration</code>
 - <code>runtimeAssertionsBuild=false</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-18041104/activerecord/1733381331123/1733390097537)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #136675 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.2.6-rc]
- #136562 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3 release-blocker]
- #136561 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3.1-rc]
- #136460 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18058396?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18058396?buildTab=artifacts#/activerecord) on master @ [21d1123dce5401b1ec580d4156db5db075074506](https://github.com/cockroachdb/cockroach/commits/21d1123dce5401b1ec580d4156db5db075074506):


```
--- FAIL: TokenForTest#test_finds_record_through_subclass (unexpected)
--- FAIL: TokenForTest#test_finds_record_when_token_has_not_expired_and_embedded_data_has_not_changed (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_double_belongs_to_destroys_both_from_either_end (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasOneAssociation#test_assignment_before_either_saved (unexpected)
--- FAIL: TokenForTest#test_returns_nil_when_record_is_not_found (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_nil (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_takes_the_same_amount_of_time_regardless_of_whether_record_is_found (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_when_password_is_correct (unexpected)
--- FAIL: SecureTokenTest#test_token_values_are_generated_for_specified_attributes_and_persisted_on_save (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_happens_only_once (unexpected)
--- FAIL: SecureTokenTest#test_token_value_not_overwritten_when_present (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_association_name_is_the_same_as_join_table_name (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_does_not_authenticate_when_password_is_incorrect (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_an_empty_string (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasOneAssociation#test_build_before_child_saved (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_has_expired (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_using_multiple_passwords (unexpected)
--- FAIL: SuppressorTest#test_suppresses_validations_on_create (unexpected)
--- FAIL: SuppressorTest#test_suppresses_create_in_callback (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasOneAssociation#test_save_fails_for_invalid_has_one (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_attribute (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_should_respect_table_alias (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_password (unexpected)
--- FAIL: SecureTokenTest#test_token_on_callback (unexpected)
--- FAIL: TokenForTest#test_finds_record_with_a_custom_primary_key (unexpected)
--- FAIL: SecureTokenTest#test_token_calls_the_setter_method (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_accepts_any_object_that_implements_to_h (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_expires_in_is_different (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_embedded_data_is_different (unexpected)
--- FAIL: SuppressorTest#test_resumes_saving_after_suppression_complete (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasOneAssociation#test_not_resaved_when_unchanged (unexpected)
--- FAIL: TokenForTest#test_raises_on_bang_when_record_is_not_found (unexpected)
--- FAIL: TokenForTest#test_raises_when_token_definition_does_not_exist (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_is_skipped_if_column_was_not_selected (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasOneAssociation#test_assignment_before_parent_saved (unexpected)
--- FAIL: TestDefaultAutosaveAssociationOnAHasOneAssociation#test_save_succeeds_for_invalid_has_one_with_validate_false (unexpected)
--- FAIL: SuppressorTest#test_suppresses_update (unexpected)
--- FAIL: TokenForTest#test_raises_when_no_primary_key_has_been_declared (unexpected)
--- FAIL: AutomaticInverseFindingTests#test_has_one_and_belongs_to_with_non_default_foreign_key_should_not_find_inverse_automatically (unexpected)
--- FAIL: TokenForTest#test_supports_JSON-serializable_embedded_data (unexpected)
--- FAIL: TokenForTest#test_finds_record_by_token (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_relation (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_invalid (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: TokenForTest#test_tokens_do_not_expire_by_default (unexpected)
--- FAIL: SecureTokenTest#test_regenerating_the_secure_token (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicLeases=leader</code>
 - <code>runtimeAssertionsBuild=false</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-18058396/activerecord/1733467722205/1733477066708)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #136675 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.2.6-rc]
- #136562 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3 release-blocker]
- #136561 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3.1-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18075244?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18075244?buildTab=artifacts#/activerecord) on master @ [9744e5f1676a752d5b200fe7bce84ca8b44afca0](https://github.com/cockroachdb/cockroach/commits/9744e5f1676a752d5b200fe7bce84ca8b44afca0):


```
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: SecureTokenTest#test_token_values_are_generated_for_specified_attributes_and_persisted_on_save (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_happens_only_once (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_does_not_authenticate_when_password_is_incorrect (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_invalid (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_expires_in_is_different (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_an_empty_string (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_has_expired (unexpected)
--- FAIL: TokenForTest#test_returns_nil_when_record_is_not_found (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_relation (unexpected)
--- FAIL: TokenForTest#test_raises_when_no_primary_key_has_been_declared (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_takes_the_same_amount_of_time_regardless_of_whether_record_is_found (unexpected)
--- FAIL: TokenForTest#test_raises_when_token_definition_does_not_exist (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_double_belongs_to_destroys_both_from_either_end (unexpected)
--- FAIL: SuppressorTest#test_suppresses_validations_on_create (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_accepts_any_object_that_implements_to_h (unexpected)
--- FAIL: TokenForTest#test_finds_record_by_token (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_for_a_different_purpose (unexpected)
--- FAIL: TokenForTest#test_raises_on_bang_when_record_is_not_found (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_finds_record_using_multiple_attributes (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: SuppressorTest#test_suppresses_create_in_callback (unexpected)
--- FAIL: SuppressorTest#test_resumes_saving_after_suppression_complete (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_attribute (unexpected)
--- FAIL: SecureTokenTest#test_token_on_callback (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_when_password_is_correct (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_is_skipped_if_column_was_not_selected (unexpected)
--- FAIL: SecureTokenTest#test_token_value_not_overwritten_when_present (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_nil (unexpected)
--- FAIL: TokenForTest#test_tokens_do_not_expire_by_default (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_password (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_association_name_is_the_same_as_join_table_name (unexpected)
--- FAIL: TokenForTest#test_finds_record_when_token_has_not_expired_and_embedded_data_has_not_changed (unexpected)
--- FAIL: SuppressorTest#test_suppresses_update (unexpected)
--- FAIL: AutomaticInverseFindingTests#test_has_one_and_belongs_to_with_non_default_foreign_key_should_not_find_inverse_automatically (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_embedded_data_is_different (unexpected)
--- FAIL: TokenForTest#test_finds_record_with_a_custom_primary_key (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_should_respect_table_alias (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_using_multiple_passwords (unexpected)
--- FAIL: SecureTokenTest#test_token_calls_the_setter_method (unexpected)
--- FAIL: TokenForTest#test_subclasses_can_redefine_tokens (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_subclass (unexpected)
--- FAIL: TokenForTest#test_supports_JSON-serializable_embedded_data (unexpected)
--- FAIL: SecureTokenTest#test_regenerating_the_secure_token (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/cpu_arch=arm64/run_1
```

Parameters:
 - <code>arch=arm64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=false</code>
 - <code>metamorphicLeases=default</code>
 - <code>runtimeAssertionsBuild=false</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-18075244/activerecord/1733554138711/1733563737162)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #136876 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.2 release-blocker]
- #136675 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.2.6-rc]
- #136562 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3 release-blocker]
- #136561 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3.1-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18085015?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18085015?buildTab=artifacts#/activerecord) on master @ [d7ea85402dc35e36c6cc35520fa91f25fd5c999d](https://github.com/cockroachdb/cockroach/commits/d7ea85402dc35e36c6cc35520fa91f25fd5c999d):


```
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_password (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_association_name_is_the_same_as_join_table_name (unexpected)
--- FAIL: SuppressorTest#test_suppresses_validations_on_create (unexpected)
--- FAIL: TokenForTest#test_tokens_do_not_expire_by_default (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_finds_record_using_multiple_attributes (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_using_multiple_passwords (unexpected)
--- FAIL: SuppressorTest#test_suppresses_create_in_callback (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_is_skipped_if_column_was_not_selected (unexpected)
--- FAIL: SecureTokenTest#test_token_on_callback (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_double_belongs_to_destroys_both_from_either_end (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_has_expired (unexpected)
--- FAIL: TokenForTest#test_supports_JSON-serializable_embedded_data (unexpected)
--- FAIL: AutomaticInverseFindingTests#test_has_one_and_belongs_to_with_non_default_foreign_key_should_not_find_inverse_automatically (unexpected)
--- FAIL: TokenForTest#test_raises_when_no_primary_key_has_been_declared (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_nil (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: SecureTokenTest#test_regenerating_the_secure_token (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_for_a_different_purpose (unexpected)
--- FAIL: SuppressorTest#test_resumes_saving_after_suppression_complete (unexpected)
--- FAIL: SecureTokenTest#test_token_calls_the_setter_method (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: TokenForTest#test_does_not_find_record_when_embedded_data_is_different (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_takes_the_same_amount_of_time_regardless_of_whether_record_is_found (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_an_empty_string (unexpected)
--- FAIL: TokenForTest#test_returns_nil_when_record_is_not_found (unexpected)
--- FAIL: TokenForTest#test_raises_when_token_definition_does_not_exist (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_relation (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_invalid (unexpected)
--- FAIL: SecureTokenTest#test_token_values_are_generated_for_specified_attributes_and_persisted_on_save (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_expires_in_is_different (unexpected)
--- FAIL: SecureTokenTest#test_token_value_not_overwritten_when_present (unexpected)
--- FAIL: TokenForTest#test_subclasses_can_redefine_tokens (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_happens_only_once (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_accepts_any_object_that_implements_to_h (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_attribute (unexpected)
--- FAIL: SuppressorTest#test_suppresses_update (unexpected)
--- FAIL: TokenForTest#test_finds_record_by_token (unexpected)
--- FAIL: TokenForTest#test_finds_record_when_token_has_not_expired_and_embedded_data_has_not_changed (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_subclass (unexpected)
--- FAIL: TokenForTest#test_finds_record_with_a_custom_primary_key (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_when_password_is_correct (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_should_respect_table_alias (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_does_not_authenticate_when_password_is_incorrect (unexpected)
--- FAIL: TokenForTest#test_raises_on_bang_when_record_is_not_found (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicLeases=default</code>
 - <code>runtimeAssertionsBuild=false</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-18085015/activerecord/1733640331320/1733647333560)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #136876 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.2 release-blocker]
- #136675 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.2.6-rc]
- #136562 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3 release-blocker]
- #136561 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3.1-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

**Note:** This build has runtime assertions enabled. If the same failure was hit in a run without assertions enabled, there should be a similar failure without this message. If there isn&#39;t one, then this failure is likely due to an assertion violation or (assertion) timeout.

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18093858?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18093858?buildTab=artifacts#/activerecord) on master @ [683618f4400ecbffb45564697fdb864909eca956](https://github.com/cockroachdb/cockroach/commits/683618f4400ecbffb45564697fdb864909eca956):


```
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_attribute (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_does_not_authenticate_when_password_is_incorrect (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_using_multiple_passwords (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_finds_record_using_multiple_attributes (unexpected)
--- FAIL: SuppressorTest#test_suppresses_validations_on_create (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_association_name_is_the_same_as_join_table_name (unexpected)
--- FAIL: SecureTokenTest#test_regenerating_the_secure_token (unexpected)
--- FAIL: TokenForTest#test_raises_when_no_primary_key_has_been_declared (unexpected)
--- FAIL: TokenForTest#test_supports_JSON-serializable_embedded_data (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_when_password_is_correct (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_password (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_invalid (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_relation (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_accepts_any_object_that_implements_to_h (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: AutomaticInverseFindingTests#test_has_one_and_belongs_to_with_non_default_foreign_key_should_not_find_inverse_automatically (unexpected)
--- FAIL: TokenForTest#test_finds_record_by_token (unexpected)
--- FAIL: SuppressorTest#test_resumes_saving_after_suppression_complete (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_an_empty_string (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_happens_only_once (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_has_expired (unexpected)
--- FAIL: SecureTokenTest#test_token_values_are_generated_for_specified_attributes_and_persisted_on_save (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_embedded_data_is_different (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_nil (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_for_a_different_purpose (unexpected)
--- FAIL: SuppressorTest#test_suppresses_create_in_callback (unexpected)
--- FAIL: SuppressorTest#test_suppresses_update (unexpected)
--- FAIL: TokenForTest#test_returns_nil_when_record_is_not_found (unexpected)
--- FAIL: TokenForTest#test_finds_record_with_a_custom_primary_key (unexpected)
--- FAIL: TokenForTest#test_finds_record_when_token_has_not_expired_and_embedded_data_has_not_changed (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_double_belongs_to_destroys_both_from_either_end (unexpected)
--- FAIL: SecureTokenTest#test_token_calls_the_setter_method (unexpected)
--- FAIL: SecureTokenTest#test_token_value_not_overwritten_when_present (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_takes_the_same_amount_of_time_regardless_of_whether_record_is_found (unexpected)
--- FAIL: TokenForTest#test_subclasses_can_redefine_tokens (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_expires_in_is_different (unexpected)
--- FAIL: TokenForTest#test_raises_on_bang_when_record_is_not_found (unexpected)
--- FAIL: TokenForTest#test_tokens_do_not_expire_by_default (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_should_respect_table_alias (unexpected)
--- FAIL: SecureTokenTest#test_token_on_callback (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_subclass (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_is_skipped_if_column_was_not_selected (unexpected)
--- FAIL: TokenForTest#test_raises_when_token_definition_does_not_exist (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicLeases=expiration</code>
 - <code>runtimeAssertionsBuild=true</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-18093858/activerecord/1733726826157/1733735696090)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #136876 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.2 release-blocker]
- #136675 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.2.6-rc]
- #136562 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3 release-blocker]
- #136561 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3.1-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

**Note:** This build has runtime assertions enabled. If the same failure was hit in a run without assertions enabled, there should be a similar failure without this message. If there isn&#39;t one, then this failure is likely due to an assertion violation or (assertion) timeout.

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18112167?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18112167?buildTab=artifacts#/activerecord) on master @ [de3b1220f5c71ac966561505c1b379060fa1407f](https://github.com/cockroachdb/cockroach/commits/de3b1220f5c71ac966561505c1b379060fa1407f):


```
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_for_a_different_purpose (unexpected)
--- FAIL: SecureTokenTest#test_token_values_are_generated_for_specified_attributes_and_persisted_on_save (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_nil (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_does_not_authenticate_when_password_is_incorrect (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_relation (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_invalid (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_password (unexpected)
--- FAIL: TokenForTest#test_raises_when_token_definition_does_not_exist (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_association_name_is_the_same_as_join_table_name (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_embedded_data_is_different (unexpected)
--- FAIL: TokenForTest#test_finds_record_by_token (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_double_belongs_to_destroys_both_from_either_end (unexpected)
--- FAIL: SecureTokenTest#test_token_calls_the_setter_method (unexpected)
--- FAIL: SecureTokenTest#test_regenerating_the_secure_token (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_should_respect_table_alias (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_happens_only_once (unexpected)
--- FAIL: SecureTokenTest#test_token_value_not_overwritten_when_present (unexpected)
--- FAIL: SuppressorTest#test_suppresses_update (unexpected)
--- FAIL: TokenForTest#test_finds_record_when_token_has_not_expired_and_embedded_data_has_not_changed (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_is_skipped_if_column_was_not_selected (unexpected)
--- FAIL: TokenForTest#test_returns_nil_when_record_is_not_found (unexpected)
--- FAIL: SuppressorTest#test_suppresses_validations_on_create (unexpected)
--- FAIL: TokenForTest#test_finds_record_with_a_custom_primary_key (unexpected)
--- FAIL: TokenForTest#test_tokens_do_not_expire_by_default (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_attribute (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_an_empty_string (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_takes_the_same_amount_of_time_regardless_of_whether_record_is_found (unexpected)
--- FAIL: TokenForTest#test_raises_when_no_primary_key_has_been_declared (unexpected)
--- FAIL: TokenForTest#test_raises_on_bang_when_record_is_not_found (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_subclass (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_finds_record_using_multiple_attributes (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_has_expired (unexpected)
--- FAIL: SuppressorTest#test_resumes_saving_after_suppression_complete (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_accepts_any_object_that_implements_to_h (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: TokenForTest#test_supports_JSON-serializable_embedded_data (unexpected)
--- FAIL: TokenForTest#test_subclasses_can_redefine_tokens (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_using_multiple_passwords (unexpected)
--- FAIL: SuppressorTest#test_suppresses_create_in_callback (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_expires_in_is_different (unexpected)
--- FAIL: AutomaticInverseFindingTests#test_has_one_and_belongs_to_with_non_default_foreign_key_should_not_find_inverse_automatically (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_when_password_is_correct (unexpected)
--- FAIL: SecureTokenTest#test_token_on_callback (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicLeases=expiration</code>
 - <code>runtimeAssertionsBuild=true</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-18112167/activerecord/1733813470860/1733820918866)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #136876 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.2 release-blocker]
- #136675 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.2.6-rc]
- #136562 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3 release-blocker]
- #136561 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3.1-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18129795?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18129795?buildTab=artifacts#/activerecord) on master @ [d18eb683b2759fd8814dacf0baa913f596074a17](https://github.com/cockroachdb/cockroach/commits/d18eb683b2759fd8814dacf0baa913f596074a17):


```
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_for_a_different_purpose (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_finds_record_using_multiple_attributes (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_is_skipped_if_column_was_not_selected (unexpected)
--- FAIL: SecureTokenTest#test_token_calls_the_setter_method (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_an_empty_string (unexpected)
--- FAIL: SuppressorTest#test_suppresses_validations_on_create (unexpected)
--- FAIL: SecureTokenTest#test_token_values_are_generated_for_specified_attributes_and_persisted_on_save (unexpected)
--- FAIL: AutomaticInverseFindingTests#test_has_one_and_belongs_to_with_non_default_foreign_key_should_not_find_inverse_automatically (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: TokenForTest#test_does_not_find_record_when_expires_in_is_different (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_password (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_relation (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_accepts_any_object_that_implements_to_h (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_double_belongs_to_destroys_both_from_either_end (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_happens_only_once (unexpected)
--- FAIL: TokenForTest#test_finds_record_when_token_has_not_expired_and_embedded_data_has_not_changed (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_takes_the_same_amount_of_time_regardless_of_whether_record_is_found (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_nil (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_has_expired (unexpected)
--- FAIL: SuppressorTest#test_resumes_saving_after_suppression_complete (unexpected)
--- FAIL: TokenForTest#test_subclasses_can_redefine_tokens (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_using_multiple_passwords (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_invalid (unexpected)
--- FAIL: SuppressorTest#test_suppresses_create_in_callback (unexpected)
--- FAIL: TokenForTest#test_raises_when_no_primary_key_has_been_declared (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_does_not_authenticate_when_password_is_incorrect (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_when_password_is_correct (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_attribute (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_embedded_data_is_different (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_association_name_is_the_same_as_join_table_name (unexpected)
--- FAIL: SecureTokenTest#test_token_on_callback (unexpected)
--- FAIL: SuppressorTest#test_suppresses_update (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_should_respect_table_alias (unexpected)
--- FAIL: TokenForTest#test_raises_on_bang_when_record_is_not_found (unexpected)
--- FAIL: TokenForTest#test_returns_nil_when_record_is_not_found (unexpected)
--- FAIL: TokenForTest#test_raises_when_token_definition_does_not_exist (unexpected)
--- FAIL: SecureTokenTest#test_token_value_not_overwritten_when_present (unexpected)
--- FAIL: TokenForTest#test_finds_record_with_a_custom_primary_key (unexpected)
--- FAIL: TokenForTest#test_supports_JSON-serializable_embedded_data (unexpected)
--- FAIL: SecureTokenTest#test_regenerating_the_secure_token (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_subclass (unexpected)
--- FAIL: TokenForTest#test_tokens_do_not_expire_by_default (unexpected)
--- FAIL: TokenForTest#test_finds_record_by_token (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicLeases=epoch</code>
 - <code>runtimeAssertionsBuild=false</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-18129795/activerecord/1733899663411/1733908372638)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #136876 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.2]
- #136675 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.2.6-rc]
- #136562 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3 release-blocker]
- #136561 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3.1-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

**Note:** This build has runtime assertions enabled. If the same failure was hit in a run without assertions enabled, there should be a similar failure without this message. If there isn&#39;t one, then this failure is likely due to an assertion violation or (assertion) timeout.

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18147340?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18147340?buildTab=artifacts#/activerecord) on master @ [e8ee6f574ddf1fce1a4cb53f392c5a9baf633b76](https://github.com/cockroachdb/cockroach/commits/e8ee6f574ddf1fce1a4cb53f392c5a9baf633b76):


```
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: TokenForTest#test_finds_record_when_token_has_not_expired_and_embedded_data_has_not_changed (unexpected)
--- FAIL: SecureTokenTest#test_token_values_are_generated_for_specified_attributes_and_persisted_on_save (unexpected)
--- FAIL: TokenForTest#test_finds_record_with_a_custom_primary_key (unexpected)
--- FAIL: TokenForTest#test_returns_nil_when_record_is_not_found (unexpected)
--- FAIL: TokenForTest#test_tokens_do_not_expire_by_default (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_embedded_data_is_different (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_double_belongs_to_destroys_both_from_either_end (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_relation (unexpected)
--- FAIL: SecureTokenTest#test_token_on_callback (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_an_empty_string (unexpected)
--- FAIL: TokenForTest#test_subclasses_can_redefine_tokens (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_finds_record_using_multiple_attributes (unexpected)
--- FAIL: SecureTokenTest#test_token_value_not_overwritten_when_present (unexpected)
--- FAIL: SecureTokenTest#test_regenerating_the_secure_token (unexpected)
--- FAIL: TokenForTest#test_finds_record_by_token (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_nil (unexpected)
--- FAIL: TokenForTest#test_raises_when_token_definition_does_not_exist (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_invalid (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_subclass (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_does_not_authenticate_when_password_is_incorrect (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_happens_only_once (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_accepts_any_object_that_implements_to_h (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_password (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_expires_in_is_different (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_attribute (unexpected)
--- FAIL: TokenForTest#test_raises_when_no_primary_key_has_been_declared (unexpected)
--- FAIL: SuppressorTest#test_suppresses_update (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_is_skipped_if_column_was_not_selected (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_takes_the_same_amount_of_time_regardless_of_whether_record_is_found (unexpected)
--- FAIL: SuppressorTest#test_suppresses_create_in_callback (unexpected)
--- FAIL: AutomaticInverseFindingTests#test_has_one_and_belongs_to_with_non_default_foreign_key_should_not_find_inverse_automatically (unexpected)
--- FAIL: SuppressorTest#test_suppresses_validations_on_create (unexpected)
--- FAIL: TokenForTest#test_supports_JSON-serializable_embedded_data (unexpected)
--- FAIL: SuppressorTest#test_resumes_saving_after_suppression_complete (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_association_name_is_the_same_as_join_table_name (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_should_respect_table_alias (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_using_multiple_passwords (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_when_password_is_correct (unexpected)
--- FAIL: TokenForTest#test_raises_on_bang_when_record_is_not_found (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_has_expired (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_for_a_different_purpose (unexpected)
--- FAIL: SecureTokenTest#test_token_calls_the_setter_method (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicLeases=leader</code>
 - <code>runtimeAssertionsBuild=true</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-18147340/activerecord/1733986069081/1733993371922)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #136876 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.2]
- #136675 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.2.6-rc]
- #136562 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3 release-blocker]
- #136561 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3.1-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18162417?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18162417?buildTab=artifacts#/activerecord) on master @ [603ff88e54d3e3f6e49b0f673abd5ec564bf418b](https://github.com/cockroachdb/cockroach/commits/603ff88e54d3e3f6e49b0f673abd5ec564bf418b):


```
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: TokenForTest#test_does_not_find_record_when_expires_in_is_different (unexpected)
--- FAIL: TokenForTest#test_finds_record_with_a_custom_primary_key (unexpected)
--- FAIL: SuppressorTest#test_suppresses_create_in_callback (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_password (unexpected)
--- FAIL: SecureTokenTest#test_regenerating_the_secure_token (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_relation (unexpected)
--- FAIL: SecureTokenTest#test_token_values_are_generated_for_specified_attributes_and_persisted_on_save (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: TokenForTest#test_returns_nil_when_record_is_not_found (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_when_password_is_correct (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_embedded_data_is_different (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_nil (unexpected)
--- FAIL: AutomaticInverseFindingTests#test_has_one_and_belongs_to_with_non_default_foreign_key_should_not_find_inverse_automatically (unexpected)
--- FAIL: TokenForTest#test_tokens_do_not_expire_by_default (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_should_respect_table_alias (unexpected)
--- FAIL: SecureTokenTest#test_token_calls_the_setter_method (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_takes_the_same_amount_of_time_regardless_of_whether_record_is_found (unexpected)
--- FAIL: SecureTokenTest#test_token_value_not_overwritten_when_present (unexpected)
--- FAIL: TokenForTest#test_supports_JSON-serializable_embedded_data (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_for_a_different_purpose (unexpected)
--- FAIL: SuppressorTest#test_suppresses_validations_on_create (unexpected)
--- FAIL: TokenForTest#test_subclasses_can_redefine_tokens (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_does_not_authenticate_when_password_is_incorrect (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_subclass (unexpected)
--- FAIL: TokenForTest#test_finds_record_when_token_has_not_expired_and_embedded_data_has_not_changed (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_has_expired (unexpected)
--- FAIL: SuppressorTest#test_suppresses_update (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_invalid (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_accepts_any_object_that_implements_to_h (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_happens_only_once (unexpected)
--- FAIL: TokenForTest#test_raises_on_bang_when_record_is_not_found (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_an_empty_string (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: TokenForTest#test_raises_when_token_definition_does_not_exist (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_attribute (unexpected)
--- FAIL: SecureTokenTest#test_token_on_callback (unexpected)
--- FAIL: TokenForTest#test_raises_when_no_primary_key_has_been_declared (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_is_skipped_if_column_was_not_selected (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_double_belongs_to_destroys_both_from_either_end (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_association_name_is_the_same_as_join_table_name (unexpected)
--- FAIL: SuppressorTest#test_resumes_saving_after_suppression_complete (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_finds_record_using_multiple_attributes (unexpected)
--- FAIL: TokenForTest#test_finds_record_by_token (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_using_multiple_passwords (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicLeases=default</code>
 - <code>runtimeAssertionsBuild=false</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-18162417/activerecord/1734072400735/1734080660529)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #136876 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.2]
- #136675 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2.6-rc]
- #136562 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3]
- #136561 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.3.1-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/cmd/roachtest/tests/activerecord.go
+++ b/pkg/cmd/roachtest/tests/activerecord.go
@@ -111,11 +111,11 @@ func registerActiveRecord(r registry.Registry) {
 			t,
 			c,
 			node,
-			"install ruby 3.1",
+			"install ruby 3.3",
 			`mkdir -p ruby-install && \
         curl -fsSL https://github.com/postmodern/ruby-install/archive/v0.9.1.tar.gz | tar --strip-components=1 -C ruby-install -xz && \
         sudo make -C ruby-install install && \
-        sudo ruby-install --system ruby 3.1.4 && \
+        sudo ruby-install --system ruby 3.3.6 && \
         sudo gem update --system`,
 		); err != nil {
 			t.Fatal(err)
@@ -146,7 +146,7 @@ func registerActiveRecord(r registry.Registry) {
 			c,
 			node,
 			"installing bundler",
-			`cd /mnt/data1/activerecord-cockroachdb-adapter/ && sudo gem install bundler:2.4.9`,
+			`cd /mnt/data1/activerecord-cockroachdb-adapter/ && sudo gem install bundler:2.5.23`,
 		); err != nil {
 			t.Fatal(err)
 		}

